draft-davidben-tls-trust-expr.md

```diff
@@ -441,39 +441,49 @@ If a relying party receives this extension in the Certificate message, it SHOULD
 
 ## Evaluating Trust Expressions
 
-Given a certification path with a `trust_store` certificate property ({{trust-store-inclusion}}), a subscriber can evaluate a TrustExpressionList to determine whether the certification path matches.
+Given a certification path and a TrustExpressionList from the relying party, the subscriber runs the following procedure to determine if the relying party is claiming support for the path's trust anchor. If so, the TrustExpressionList is said to match the certification path:
 
-A TrustExpression is said to match a TrustStoreInclusionList if there is at least one TrustStoreInclusion in the TrustStoreInclusionList such that all of the following are true:
+1. If the certification path does not have an associated `trust_store` certificate property ({{trust-store-inclusion}}), containing a TrustStoreInclusionList, terminate this procedure and return false. The certification path does not match the TrustExpressionList.
 
-* The `name` fields of two structures' `trust_store` fields are equal.
+2. If the certification path has expired, terminate this procedure and return false. Expired certification paths are expected to be rejected by the relying party and may also have stale TrustStoreInclusionLists.
```

Comment:
I don't think we want to do this here.  If for (reasons) my certificates have expired, today, I still serve up the expired certificate rather than effectively, failing. It is then up to the relying party to decide what to do with an expired certificate and this changes that dynamic. 

I think an expired path is *still* a match.  If there is *more than one* match and one is expired and one is not, then absolutely, choose the non-expired one to serve.

Reply:
Consider a future PQ world with PQ clients, and a default classical cert - it's worse behaviour to serve up the classical cert because the PQ cert  that matched the expression was expired, than to just serve up the expired cert we know matches because it's the best we have.

Reply:
The problem is that, if expired, all the version skew machinery we've put in no longer works. The relying party only needs to send `excluded_labels` up to when the old certificates expire. If we say that expired certificates may match, then the relying party can never retire `excluded_labels` and then removals leave persistent scar tissue.

Reply:
After conversation offline, I'm maybe kinda ok with this, as long as we call out the new dependency on the server having accurate time, which I think we need to note, as well as the differences here that we would noa serve the default cert in this case instead of the expired one in some situations.